src/griffe/git.py

```diff
@@ -0,0 +1,120 @@
+"""This module contains the code allowing to load modules from specific git commits.
+
+```python
+from griffe.git import load_git
+
+# where `repo` is the folder *containing* `.git`
+old_api = load_git("my_module", commit="v0.1.0", repo="path/to/repo")
+```
+"""
+from __future__ import annotations
+
+import os
+import shutil
+from contextlib import contextmanager
+from pathlib import Path
+from subprocess import DEVNULL, PIPE, CalledProcessError, check_output, run  # noqa: S404
+from tempfile import TemporaryDirectory
+from typing import Any, Iterator, Sequence
+from uuid import uuid1
+
+from griffe import loader
+from griffe.agents.extensions import Extensions
+from griffe.collections import LinesCollection, ModulesCollection
+from griffe.dataclasses import Module
+from griffe.docstrings.parsers import Parser
+
+
+def _assert_git_repo(repo: str) -> None:
+    if not shutil.which("git"):
+        raise RuntimeError("Could not find git executable. Please install git.")
+
+    try:
+        check_output(["git", "-C", repo, "rev-parse", "--is-inside-work-tree"], stderr=DEVNULL)  # noqa: S603,S607
```

Comment:
I think you can use `run` with `check=True` instead of `check_output`, that would be one less import. What do you think? Maybe I'm missing something.